lightning/src/ln/features.rs

```diff
@@ -337,68 +529,88 @@ impl<T: sealed::Context> Readable for Features<T> {
+		assert!(!InitFeatures::known().requires_payment_secret());
+		assert!(!NodeFeatures::known().requires_payment_secret());
+
+		assert!(InitFeatures::known().supports_basic_mpp());
+		assert!(NodeFeatures::known().supports_basic_mpp());
+		assert!(!InitFeatures::known().requires_basic_mpp());
+		assert!(!NodeFeatures::known().requires_basic_mpp());
+
+		let mut init_features = InitFeatures::known();
+		assert!(init_features.initial_routing_sync());
+		init_features.clear_initial_routing_sync();
+		assert!(!init_features.initial_routing_sync());
 	}
 
 	#[test]
-	fn sanity_test_unkown_bits_testing() {
-		let mut features = ChannelFeatures::supported();
-		features.set_require_unknown_bits();
+	fn sanity_test_unknown_bits() {
+		let mut features = ChannelFeatures::empty();
+		assert!(!features.requires_unknown_bits());
+		assert!(!features.supports_unknown_bits());
+
+		features.set_required_unknown_bits();
 		assert!(features.requires_unknown_bits());
-		features.clear_require_unknown_bits();
+		assert!(features.supports_unknown_bits());
+
+		features.clear_unknown_bits();
+		assert!(!features.requires_unknown_bits());
+		assert!(!features.supports_unknown_bits());
+
+		features.set_optional_unknown_bits();
 		assert!(!features.requires_unknown_bits());
+		assert!(features.supports_unknown_bits());
 	}
 
 	#[test]
-	fn test_node_with_known_relevant_init_flags() {
-		// Create an InitFeatures with initial_routing_sync supported.
-		let mut init_features = InitFeatures::supported();
-		init_features.set_initial_routing_sync();
-
-		// Attempt to pull out non-node-context feature flags from these InitFeatures.
-		let res = NodeFeatures::with_known_relevant_init_flags(&init_features);
+	fn convert_to_context_with_relevant_flags() {
+		let init_features = InitFeatures::known().clear_upfront_shutdown_script();
+		assert!(init_features.initial_routing_sync());
+		assert!(!init_features.supports_upfront_shutdown_script());
 
+		let node_features: NodeFeatures = init_features.to_context();
 		{
-			// Check that the flags are as expected: optional_data_loss_protect,
-			// option_upfront_shutdown_script, var_onion_optin, payment_secret, and
-			// basic_mpp.
-			assert_eq!(res.flags.len(), 3);
-			assert_eq!(res.flags[0], 0b00100010);
-			assert_eq!(res.flags[1], 0b10000010);
-			assert_eq!(res.flags[2], 0b00000010);
+			// Check that the flags are as expected:
+			// - option_data_loss_protect
+			// - var_onion_optin | payment_secret
+			// - basic_mpp
+			assert_eq!(node_features.flags.len(), 3);
+			assert_eq!(node_features.flags[0], 0b00000010);
+			assert_eq!(node_features.flags[1], 0b10000010);
+			assert_eq!(node_features.flags[2], 0b00000010);
 		}
 
-		// Check that the initial_routing_sync feature was correctly blanked out.
-		let new_features: InitFeatures = Features::from_le_bytes(res.flags);
-		assert!(!new_features.initial_routing_sync());
+		// Check that cleared flags are kept blank when converting back.
```

Comment:
" - initial_routing_sync can't be part of node context and its feature flag should always be cleared for a conversion from node
- upfront_shutdown_script was cleared before context conversion and should still be cleared after converting back